billing/views.py

```diff
@@ -54,38 +53,40 @@ def invoice_payment_failed(self, invoice: stripe.Invoice) -> None:
                 stripe_subscription_id=invoice.subscription,
             ),
         )
-        updated = Owner.objects.filter(
+        updated: int = Owner.objects.filter(
             stripe_customer_id=invoice.customer,
             stripe_subscription_id=invoice.subscription,
         ).update(delinquent=True)
         self._log_updated(updated)
 
     def customer_subscription_deleted(self, subscription: stripe.Subscription) -> None:
-        try:
+        log.info(
+            "Customer Subscription Deleted - Setting free plan and deactivating repos for stripe customer",
+            extra=dict(
+                stripe_subscription_id=subscription.id,
+                stripe_customer_id=subscription.customer,
+            ),
+        )
+        owners: QuerySet[Owner] = Owner.objects.filter(
+            stripe_customer_id=subscription.customer,
+            stripe_subscription_id=subscription.id,
+        )
+        if not owners.exists():
             log.info(
-                "Customer Subscription Deleted - Setting free plan and deactivating repos for stripe customer",
+                "Customer Subscription Deleted - Couldn't find owner, subscription likely already deleted",
                 extra=dict(
                     stripe_subscription_id=subscription.id,
                     stripe_customer_id=subscription.customer,
                 ),
             )
-            owner: Owner = Owner.objects.get(
-                stripe_customer_id=subscription.customer,
-                stripe_subscription_id=subscription.id,
-            )
-            plan_service = PlanService(current_org=owner)
+            return
+
+        for owner in owners:
+            plan_service = PlanService(owner)
             plan_service.set_default_plan_data()
             owner.repository_set.update(active=False, activated=False)
 
-            self._log_updated(1)
-        except Owner.DoesNotExist:
-            log.info(
-                "Customer Subscription Deleted - Couldn't find owner, subscription likely already deleted",
-                extra=dict(
-                    stripe_subscription_id=subscription.id,
-                    stripe_customer_id=subscription.customer,
-                ),
-            )
+        self._log_updated(owners.count())
```

Comment:
Since we're adding this log here now, what are your thoughts on improving it further by sending the owner_ids into the log itself?

Reply:
Great idea I'll do this!